pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,16 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
+				return errors.New("resourceVersion cannot be set on objects store in etcd")
+			}
+		}
+		// TODO: If versioner is nil, then we may end up with having ResourceVersion set
+		// in the object and this will be incorrect ResourceVersion. We should fix it by
+		// requiring "versioner != nil" at the constructor level for 1.3 milestone.
+
```

Comment:
@smarterclayton - I browsed through the code, and basically there is no easy way to construct etcdHelper with nil versioner (all constructors set it to ApiObjectVersioner{}) and we don't have setters to change it.
So what do you think about fixing this TODO in a followup PR now?